src/braille/ibraille.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2023 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef MU_BRAILLE_IBRAILLE_H
+#define MU_BRAILLE_IBRAILLE_H
+
+#include <QIODevice>
+
+#include "modularity/imoduleexport.h"
+
+namespace mu::engraving {
+class Score;
+}
+
+namespace mu::braille {
+class IBraille;
+using IBraillePtr = std::shared_ptr<IBraille>;
+
+class IBraille : MODULE_EXPORT_INTERFACE
```

Comment:
I think better rename to like `IBrailleWriter`

Reply:
@igorkorsukov, sure. How about this as well:

- Rename `Braille` to `BrailleWriter`.
- Rename `BrailleImpl` to `Braille`.

And then define the `BrailleWriter` and `Braille` classes in separate files in `src/braille/internal`, instead of in the same file like they are now?

Reply:
@igorkorsukov, I decided to call it `IBrailleConverter` instead of `IBrailleWriter` because it doesn't inherit from `INotationWriter`. Also, in the future it will contain new methods that don't write to a device but instead create lists of type `BrailleEngravingItem`. It's probably better to describe that as "converting" rather the "writing".

Other changes since your last review:
- Renamed `Braille` class to `BrailleConverter`.
- Renamed `BrailleImpl` class to `Braille`.
- Moved `Braille` and `TextToUEBBraille` classes to a new file.
- Rearranged methods within the `Braille` class to put them in alphabetical order and ensure the order in the `.cpp` source file matches the order in the `.h` header.